test/app/models/Interactive.spec.js

```diff
@@ -0,0 +1,269 @@
+
+import Ajv from 'ajv'
+
+// If we want to use both draft-04 and draft-06/07 schemas then use { schemaId: 'auto' }
+const ajv = new Ajv()
+
+function schemaCompileTest (schemaObject) {
+  it('compiles successfully', () => {
+    const validate = ajv.compile(schemaObject)
+    expect(typeof validate).toBe('function')
+    expect(ajv.errors).toBe(null)
+  })
+}
+
+function schemaValidateObjectTest (schemaObject, testObject) {
+  it('validates a correct object', () => {
+    const valid = ajv.validate(schemaObject, testObject)
+    expect(valid).toBe(true)
+    expect(ajv.errors).toBe(null)
+  })
+}
+
+function schemaValidateBadObjectTest (schemaObject, testObject) {
+  it('fails to validate an incorrect object', () => {
+    const valid = ajv.validate(schemaObject, testObject)
+    expect(valid).toBe(false)
+    expect(ajv.errors).toBeDefined()
+  })
+}
+
+function schemaValidateBadPropertyTest (schemaObject, testObject) {
+  it('fails to validate incorrect properties on object', () => {
+    const valid = ajv.validate(schemaObject, testObject)
+    expect(valid).toBe(false)
+    expect(ajv.errors).toBeDefined()
+  })
+}
+
+const interactiveDraggableObject = {
+  labels: ['label1', 'label2'],
+  elements: [{
+    text: 'element 1',
+    elementId: '0123456789abcdefghijklmn'
+  }, {
+    text: 'element 2',
+    elementId: '1123456789abcdefghijklmn'
+  }],
+  solution: ['1123456789abcdefghijklmn', '0123456789abcdefghijklmn']
+}
+
+const interactiveInsertCodeObject = {
+  starterCode: 'run ',
+  choices: [{
+    text: 'AWAY!!!',
+    choiceId: '0123456789abcdefghijklmn',
+    triggerArt: 'running-away'
+  }],
+  solution: '0123456789abcdefghijklmn'
+}
+
+const interactiveDraggableClassificationObject = {
+  categories: [{
+    categoryId: 'c123456789abcdefghijklmn',
+    text: 'Something draggable'
+  }],
+  elements: [{
+    text: 'Draggable number one',
+    elementId: '0123456789abcdefghijklmn'
+  }],
+  solution: [{
+    categoryId: 'c123456789abcdefghijklmn',
+    elements: ['0123456789abcdefghijklmn']
+  }]
+}
+
+const interactiveMultipleChoiceObject = {
+  choices: [{
+    text: 'It is dangerous to go alone',
+    choiceId: '0123456789abcdefghijklmn'
+  }],
+  solution: '0123456789abcdefghijklmn'
+}
+
+const interactiveFillInCodeObject = {
+  starterCode: 'console.log("hello world!")',
+  commonResponses: [{
+    text: 'Hi world',
+    responseId: '0123456789abcdefghijklmn',
+    triggerArt: 'globe'
+  }],
+  solution: '0123456789abcdefghijklmn'
+}
+
+const interactiveDraggableStatementCompletionObject = {
+  labels: ['hello', 'world'],
+  elements: [{
+    text: 'world',
+    elementId: '0123456789abcdefghijklmn'
+  }, {
+    text: 'hello',
+    elementId: '1123456789abcdefghijklmn'
+  }],
+  solution: ['1123456789abcdefghijklmn', '0123456789abcdefghijklmn']
+}
+
+describe('interactive solutions schema tests', () => {
+  beforeEach(() => {
+    ajv.errors = null
```

Comment:
Open question: As we are sometimes having issues with `ajv.errors` still being defined from the last run, is it better to reset errors or to recreate a new `ajv` for each run? 

In general I feel hesitation when having to work around the way a package behaves like this. It makes me think that we'd always want to recreate `ajv` for each use, and never re-use an instance of it across the codebase like we do with `GLOBAL.tv4`. 

Thoughts?

Reply:
I think I would prefer to re create a fresh one every time rather than track a global.

Reply:
Updating to create a new instance in the re-used functions like `schemaCompileTest`, so we dont need to clear the errors every time.